src/components/DailyRoutinePage/DailyRoutine.js

```diff
@@ -1,35 +1,84 @@
 import "./DailyRoutine.css";
 import AddedProductCard from "./AddedProductCard.js";
 import { FaArrowLeft } from "react-icons/fa";
+import { useParams, useHistory } from "react-router-dom";
+import { getDataFromLocalStorage } from "../../utility/localStorage.js";
+import { useEffect, useState } from "react";
 
 export default function DailyRoutine() {
+  const { weekday } = useParams();
+  const history = useHistory();
+  const [allItems, setAllItems] = useState([]);
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    const fetchProducts = () => {
+      return fetch("/products.json")
+        .then((response) => response.json())
+        .then((productData) => {
+          setProducts(productData);
+        });
+    };
+    fetchProducts();
+  }, []);
+
+  useEffect(() => {
+    const routine = getDataFromLocalStorage();
+    setAllItems(routine);
+  }, [weekday]);
+
+  let i;
+  if (weekday === "Monday") {
+    i = 0;
+  } else if (weekday === "Tuesday") {
+    i = 1;
+  } else if (weekday === "Wednesday") {
+    i = 2;
+  } else if (weekday === "Thursday") {
+    i = 3;
+  } else if (weekday === "Friday") {
+    i = 4;
+  } else if (weekday === "Saturday") {
+    i = 5;
+  } else if (weekday === "Sunday") {
+    i = 6;
+  }
```

Comment:
It seems you are doing this in many places. Why don't you create a utility function (something like `getIndexForWeekDay(weekDay)` that you can use everywhere?